pkg/collect/cluster_resources_test.go

```diff
@@ -1,14 +1,200 @@
 package collect
 
 import (
+	"context"
 	"reflect"
 	"testing"
 
 	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	testclient "k8s.io/client-go/kubernetes/fake"
 )
 
+func Test_ConfigMaps(t *testing.T) {
+	tests := []struct {
+		name          string
+		configMapName string
+		namespaces    []string
+	}{
+		{
+			name:          "single namespace",
+			configMapName: "default",
+			namespaces:    []string{"default"},
+		},
+		{
+			name:          "multiple namespaces",
+			configMapName: "default",
+			namespaces:    []string{"default", "test"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := testclient.NewSimpleClientset()
+			ctx := context.Background()
+			err := createConfigMaps(client, tt.configMapName, tt.namespaces)
+			require.NoError(t, err)
+
+			configMaps, _ := configMaps(ctx, client, tt.namespaces)
+			assert.Equal(t, len(tt.namespaces), len(configMaps))
+
+			for _, ns := range tt.namespaces {
+				assert.NotEmpty(t, configMaps[ns+".json"])
+			}
+		})
+	}
+}
+
+func createConfigMaps(client kubernetes.Interface, configMapName string, namespaces []string) error {
+	for _, ns := range namespaces {
+		_, err := client.CoreV1().ConfigMaps(ns).Create(context.Background(), &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: configMapName,
+			},
+		}, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Test_VolumeAttachments(t *testing.T) {
+	tests := []struct {
+		name      string
+		leaseName string
+	}{
+		{
+			name:      "single namespace",
+			leaseName: "default",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := testclient.NewSimpleClientset()
+			ctx := context.Background()
+			err := createTestVolumeAttachments(client, tt.leaseName)
+			require.NoError(t, err)
+
+			volumeAttachments, _ := volumeAttachments(ctx, client)
+			assert.NotEmpty(t, volumeAttachments)
```

Comment:
do we want to check that the volume attachment name exists in the JSON...? at least checking 1 element would give confidence its valid JSON